### DISCUSSION TITLE
`@site` and `@theme` in MDX file cannot be properly index on WebStorm

### DISCUSSION THREAD

**Discussion by tisonkun** [facebook/docusaurus#8187] [Q&A] [answered]
<img width="910" alt="image" src="https://user-images.githubusercontent.com/18818196/194492597-abce4f2a-8143-4501-a109-2711500b668a.png">

<img width="718" alt="image" src="https://user-images.githubusercontent.com/18818196/194492661-eb602aaf-ee23-44ab-88c1-e04349427612.png">

How can I get it properly indexed?

**Comment by slorber** [comment, marked as answer]
Hey, unfortunately it is out of the scope of Docusaurus and more a possible limitation of your IDE.

See for example 
https://stackoverflow.com/questions/34943631/path-aliases-for-imports-in-webstorm
https://intellij-support.jetbrains.com/hc/en-us/community/posts/360003361399-Webstorm-does-not-see-alias

Some suggest to create a `webStorm.config.js` (can't find any official doc on it) or `webpack.config.js`



In my case, I have IDEA Intellij (similarto WebStorm) and it seems to work thanks to the TypeScript support: our base Docusaurus config defines the webpack `@site` alias and Intellij seems to read/understand it properly

**Reply by tisonkun** [reply]
Agree. It seems WebStorm MDX plugin's fault. In the source `ts`/`tsx` files it can be properly resolved in my env.

Although, `import {useDocsSidebar} from '@docusaurus/theme-common/internal'` complains with module not found ... too many troubleshooting to integrate with IDE I'd say :)

**Reply by slorber** [reply]
Yes this is always a pain 😓 Had to deal with similar issues many times with JetBrains IDEs, they are powerful but often a little detail ruins the experience

Not even sure WebStorm understand very well the "exports" field which is relatively new to package.json

![CleanShot 2022-10-07 at 12 48 25@2x](https://user-images.githubusercontent.com/749374/194536568-1d64031e-0167-48a9-805f-4ccf2c428973.png)
